scene_text_recognition/README.en.md

```diff
@@ -0,0 +1,141 @@
+  --help                       Show this message and exit.
+
+\`\`\`
+
+- `train_file_list` ：The list file of training data, which contain file path and corresponding  text label for each row.
+\`\`\`
+word\_1.png, "PROPER"
+word\_2.png, "FOOD"
+\`\`\`
+- `test_file_list` ：The list file of the test data .The format is the  same as above.
+- `label_dict_path` ：The path of the lable  in the training data. If the dictionary file does not exist in the specified path, the program will automatically generate the markup dictionary using the tagged data from the training data.
+- `model_save_dir` ：The saved path of model parameters .`./models` by default.
+
+### The process of specific implementation:
+
+1.Download the data [[2][4]] (Task 2.3: Word Recognition (2013 Edition) from the official website), and there will be three files: `Challenge2_Training_Task3_Images_GT.zip`, `Challenge2_Test_Task3_Images.zip`, and `Challenge2_Test_Task3_GT.txt`.
+Corresponding to the training set, the corresponding words, the pictures of the test set and the words corresponding to the test data correspond to the pictures of the training set. Then perform the following commands to extract and move the data to the target folder.
+
+\`\`\`bash
+mkdir -p data/train\_data
+mkdir -p data/test\_data
+unzip Challenge2\_Training\_Task3\_Images\_GT.zip -d data/train\_data
+unzip Challenge2\_Test\_Task3\_Images.zip -d data/test\_data
+mv Challenge2\_Test\_Task3\_GT.txt data/test\_data
+\`\`\`
+
+2.Get the path of the `gt.txt` in the training data folder (data/train\_data) and the path of the `Challenge2_Test_Task3_GT.txt` in the test data folder (data/test\_data).
+
+3.Carry out the following orders for training:
+\`\`\`bash
+python train.py \\
+--train\_file\_list\_path 'data/train\_data/gt.txt' \\
+--test\_file\_list\_path 'data/test\_data/Challenge2\_Test\_Task3\_GT.txt' \\
+--label\_dict\_path 'label\_dict.txt'
+\`\`\`
+4.During training, the model parameters are automatically backed up to the specified directory, which is stored in the `./models` directory by default.
+
+
+### Forecast
+The prediction is partly done by `infer.py`, and it uses the best path decoding algorithm, that is to select a maximum probability character at every time step. In use, you need to specify a specific model save path, a picture fixed size, a batch\_size (default 10), a dictionary path, and a list file of the picture file in the `infer.py`. The following code is executed:
+\`\`\`bash
+python infer.py \\
+--model\_path 'models/params\_pass\_00000.tar.gz' \\
+--image\_shape '173,46' \\
+--label\_dict\_path 'label\_dict.txt' \\
+--infer\_file\_list\_path 'data/test\_data/Challenge2\_Test\_Task3\_GT.txt'
+\`\`\`
+
+
+### Other datasets
+
+-   [SynthText in the Wild Dataset][5](41G)
+-   [ICDAR 2003 Robust Reading Competitions][6]
+
+### Matters needing attention
+
+- Because the `warp CTC` of the model relies on the implementation of CUDA, this model only supports GPU operation.
```

Comment:
Our model has to be runned by GPU since it relies on warp CTC, which can only be implemented by CUDA.